File: src/python/Concat1.py
#
# Concat1.py
#
# Created by Jukka Aro on 13.3.2012.
#

import sys

def concatenate(input_file, dry_run):
	f = open(input_file, 'r')
	if dry_run:
		for line in f:
			pass
	else:
		s = '';
		for line in f:
			s += line
	f.close()


# program entry:

if len(sys.argv) < 2 or len(sys.argv) > 3:
	print "Illegal number of arguments"
	sys.exit()
if len(sys.argv)  == 3 and sys.argv[2] == '--dry-run':
	dry_run = True
else:
	dry_run = False
	
concatenate(sys.argv[1], dry_run)

File: src/perl/Concat1.pl
#
# Concat1.pl
#
# Created by Jukka Aro on 14.3.2012.
#

sub concatenate{
	open FILE, $_[0] or die $!;
	if($_[1] == 0) {
		$res = "";
		while(<FILE>){
			$res .= $_;
		}
	}
	else {
		while(<FILE>){}
	}
	close(FILE);
}

#program entry:

if($#ARGV > 1){
	die "Illegal number of arguments";
}
if($#ARGV == 1 && $ARGV[1] == "--dry-run"){
	$dry_run = 1;
}
else {
	$dry_run = 0;
}
concatenate($ARGV[0], $dry_run);
